### PR TITLE
dev-lang/solidity: New package

### PR DESCRIPTION
**Pull request by MathyV**
Solidity is a high-level language whose syntax is similar
to that of JavaScript and it is designed to compile
to code for the Ethereum Virtual Machine.

There are some issues with the release tarball which have been reported upstream here:

https://github.com/ethereum/solidity/issues/1183

Package-Manager: portage-2.3.0

### CODE REVIEW

**Comment by MathyV** [comment]
I was not sure of the category but figured since it is also a compiler I'd put it next to the other compilers.

**Comment by monsieurp** [comment]
Hi @MathyV. Thank you for your contribution. Could you please also file a bug at https://bugs.gentoo.org? We want to make sure we've got your email address in your system so that bug reports can be assigned to you.

**Comment by SoapGentoo** [comment]
In addition, can we park this somewhere else? `sys-devel` is making it seem as if this thing is really a core system package. What about `dev-lang`?

File: sys-devel/solidity/solidity-0.4.2.ebuild [hunk 1]
```diff
@@ -0,0 +1,41 @@
+# Copyright 1999-2016 Gentoo Foundation
+# Distributed under the terms of the GNU General Public License v2
+# $Id$
+
+EAPI=6
+
+CMAKE_MIN_VERSION="3.0.0"
```
Comment (SoapGentoo) [inline sys-devel/solidity/solidity-0.4.2.ebuild]: is this check really important? I mean we still have 2.8 in the tree, but 3.5.2 is already stable...

File: sys-devel/solidity/solidity-0.4.2.ebuild [hunk 2]
```diff
@@ -0,0 +1,41 @@
...
+HOMEPAGE="https://github.com/ethereum/solidity"
+SRC_URI="https://github.com/ethereum/${PN}/archive/v${PV}.tar.gz -> ${P}.tar.gz"
+
+LICENSE="GPL-3"
+SLOT="0"
+KEYWORDS="~amd64 ~x86"
+IUSE=""
+
+DEPEND="dev-libs/jsoncpp:0=
+	dev-libs/boost:0=
```
Comment (SoapGentoo) [inline sys-devel/solidity/solidity-0.4.2.ebuild]: `:=` not `:0=` for both deps

File: sys-devel/solidity/solidity-0.4.2.ebuild [hunk 3]
```diff
@@ -0,0 +1,41 @@
...
+
+src_unpack() {
+	default
+
+	# The build won't work without this file but it is missing from
+	# the release tarball.
+	#
+	# Reported upstream: https://github.com/ethereum/solidity/issues/1183
+
+	cp "${FILESDIR}"/${P}-commit_hash.txt "${S}"/commit_hash.txt
```
Comment (SoapGentoo) [inline sys-devel/solidity/solidity-0.4.2.ebuild]: missing `|| die`

File: sys-devel/solidity/solidity-0.4.2.ebuild [hunk 4]
```diff
@@ -0,0 +1,41 @@
...
+
+	# Without this file the build is marked as a developer version
+	# but it is missing from the release tarball.
+	#
+	# Reported upstream: https://github.com/ethereum/solidity/issues/1183
+
+	touch "${S}"/prerelease.txt
```
Comment (SoapGentoo) [inline sys-devel/solidity/solidity-0.4.2.ebuild]: missing `|| die`

File: sys-devel/solidity/solidity-0.4.2.ebuild [hunk 5]
```diff
@@ -0,0 +1,41 @@
...
+SLOT="0"
+KEYWORDS="~amd64 ~x86"
+IUSE=""
+
+DEPEND="dev-libs/jsoncpp:0=
+	dev-libs/boost:0=
+"
+RDEPEND="${DEPEND}"
+
+src_unpack() {
```
Comment (SoapGentoo) [inline sys-devel/solidity/solidity-0.4.2.ebuild]: I suggest doing this in `src_prepare` instead

**Comment by MathyV** [comment]
@monsieurp the bug was already created ;-)

https://bugs.gentoo.org/show_bug.cgi?id=596580

File: sys-devel/solidity/solidity-0.4.2.ebuild (hunk 1 above)
Comment (MathyV) [inline sys-devel/solidity/solidity-0.4.2.ebuild]: I got it from the CMakeLists.txt file, I was asked in the other ebuild to be as precise as possible with version requirements, so I did it here too :-)

File: sys-devel/solidity/solidity-0.4.2.ebuild (hunk 2 above)
Comment (MathyV) [inline sys-devel/solidity/solidity-0.4.2.ebuild]: Fixed

File: sys-devel/solidity/solidity-0.4.2.ebuild (hunk 5 above)
Comment (MathyV) [inline sys-devel/solidity/solidity-0.4.2.ebuild]: Fixed

File: sys-devel/solidity/solidity-0.4.2.ebuild (hunk 3 above)
Comment (MathyV) [inline sys-devel/solidity/solidity-0.4.2.ebuild]: Fixed

File: sys-devel/solidity/solidity-0.4.2.ebuild (hunk 4 above)
Comment (MathyV) [inline sys-devel/solidity/solidity-0.4.2.ebuild]: Fixed

File: sys-devel/solidity/solidity-0.4.2.ebuild (hunk 5 above)
Comment (MathyV) [inline sys-devel/solidity/solidity-0.4.2.ebuild]: Fixed

File: sys-devel/solidity/solidity-0.4.2.ebuild (hunk 3 above)
Comment (MathyV) [inline sys-devel/solidity/solidity-0.4.2.ebuild]: Fixed

File: sys-devel/solidity/solidity-0.4.2.ebuild (hunk 4 above)
Comment (MathyV) [inline sys-devel/solidity/solidity-0.4.2.ebuild]: Fixed

**Comment by MathyV** [comment]
All requested changes implemented, and moved package to dev-lang/solidity

**Comment by monsieurp** [comment]
dev-util?

**Pull request closed without merging**